Sources/Sentry/SentryFileManager.m

```diff
@@ -20,19 +20,45 @@
 
 NSString *const EnvelopesPathComponent = @"envelopes";
 
+BOOL
+isErrorPathTooLong(NSError *error)
+{
+    NSError *underlyingError = NULL;
```

Comment:
@philipphofmann this should now correctly unwrap the underlying error, otherwise it will fall back to the error itself

Reply:
Hmm there is so much logic in this logging code that I would feel more comfortable adding tests for this. We can test the log output, but here I'm mostly worried about test coverage. We don't need to validate the log output, but having 100% test coverage ensuring we're not crashing would be good. We had issues in the past in logging code that led to crashes because we didn't add tests for it. Then, when an edge case occurred, the log message crashed the app.

Reply:
I agree, I'll add tests